scripts/scons_helpers/build-3rdparty.py

```diff
@@ -664,18 +664,164 @@ def splitenv(envlist):
         env[k] = v
     return env
 
+# Guess platform argument for OpenSSL's Configure script basing on toolchain string.
+# (see `./Configure LIST` for a list of all platforms and their format)
+# See also openssl/Configurations/10-main.conf
+def openssl_get_platform(toolchain):
+    """
+    >>> openssl_get_platform('x86_64-pc-linux-gnu')
+    'linux-x86_64'
+    >>> openssl_get_platform('aarch64-linux-gnu')
+    'linux-aarch64'
+    >>> openssl_get_platform('mips-openwrt-linux-uclibc')
+    'linux-mips32'
+    >>> openssl_get_platform('arm-linux-gnueabihf')
+    'linux-generic32'
+    >>> openssl_get_platform('arm-bcm2708hardfp-linux-gnueabi')
+    'linux-generic32'
+    >>> openssl_get_platform('mips64-linux-musl')
+    'linux64-mips64'
+    >>> openssl_get_platform('arm-linux-androideabi')
+    'android-arm'
+    >>> openssl_get_platform('aarch64-linux-android')
+    'android64-aarch64'
+    >>> openssl_get_platform('i686-linux-android')
+    'android-x86'
+    >>> openssl_get_platform('x86_64-linux-android')
+    'android64-x86_64'
+    >>> openssl_get_platform('i686-pc-linux-gnu')
+    'linux-x86'
+
+    # other edge cases
+    >>> openssl_get_platform('android64')
+    'android64'
+    >>> openssl_get_platform('')
+    'cc'
+    """
+
+    # we assume that:
+    # - 'android' has more priority than 'linux'
+    # - if toolchain contains '64' and 'android' => we have android64 on 64-bit CPU
+    # - if toolchain contains '64' and 'linux' => we have linux64 on 64-bit CPU
+
+    bitness = 64 if '64' in toolchain else 32
+
+    # normalize os: make toolchain to have at most 3 components
+    toolchain = toolchain.\
+        replace('bcm2708hardfp-linux', 'linux').\
+        replace('openwrt-linux', 'linux').\
+        replace('pc-linux', 'linux')
+    # normalize arch: no suffix means 32 bit
+    toolchain = toolchain.replace('mips-', 'mips32-')
+
+    ### corner cases
+
+    # 64 bit => explicitly prefer linux64 or android64 (concider libc and other libs
+    # to be 64 bit)
+    if bitness == 64:
+        if 'linux-' in toolchain:
```

Comment:
I suggest to replace `linux-` with `linux`. I can't imagine false positives for this, and it looks a safer heuristic.